src/wallet/rpcwallet.cpp

```diff
@@ -183,20 +183,20 @@ UniValue getaddressinfo(const UniValue& params, bool fHelp)
     LOCK(pwallet->cs_wallet);
 
     UniValue ret(UniValue::VOBJ);
-    CBitcoinAddress address(params[0].get_str());
+
     // Make sure the destination is valid
-    if (!address.IsValid()) {
+    if (!IsValidDestinationString(params[0].get_str())) {
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
     }
-    CTxDestination dest = address.Get();
+    CTxDestination dest = DecodeDestination(params[0].get_str());
```

Comment:
Here too. We should avoid double decoding by *first* decoding, then checking the validity on CTxDestination directly